src/itemButtonList.js

```diff
@@ -0,0 +1,226 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 Open Assessment Technologies SA ;
+ */
+import $ from 'jquery';
+import autoscroll from 'ui/autoscroll';
+import componentFactory from 'ui/component';
+import itemButtonListTpl from 'ui/itemButtonList/tpl/itemButtonList';
+import 'ui/itemButtonList/css/item-button-list.css';
+
+/**
+ * CSS classes involved in the component
+ * @type {Object}
+ */
+const cssClasses = {
+    active: 'buttonlist-item-active',
+    keyfocused: 'buttonlist-btn-focus'
+};
+
+/**
+ * CSS selectors that match some particular elements
+ * @type {Object}
+ */
+const cssSelectors = {
+    active: `.${cssClasses.active}`,
+    keyfocused: `.${cssClasses.keyfocused}`,
+    navigable: '.buttonlist-btn',
+    itemById: (id) => `.buttonlist-item[data-id="${id}"]`,
+    navigableById: (id) => `.buttonlist-btn[data-id="${id}"]`
+};
+
+/**
+ * @typedef {Object} ItemButton
+ * @property {String} id - item id
+ * @property {Number} position - 0-based list index
+ * @property {String} label - displayed text
+ * @property {String} numericLabel - displayed number (alternative to label)
+ * @property {String} ariaLabel
+ * @property {Number} score - the item's current score
+ * @property {Number} maxScore - the item's max possible score
+ * @property {Boolean} informational
+ * @property {Boolean} skipped
+ * @property {String} type - 'correct'/'incorrect'/'info'/'skipped'/'default'
+ * @property {String} status - 'answered'/'viewed'/'unseen'
+ * @property {String} scoreType - 'correct'/'incorrect'/null
+ * @property {String} icon - 'info' or null
```

Comment:
I think not all properties are relevant? Are `score/maxScore/informational/skipped/type` (maybe smth else) used?

Reply:
You are right, I've copy-pasted the typedef. It makes more sense to only document what is used here.